var/spack/repos/builtin/packages/adios/package.py

```diff
@@ -12,30 +36,97 @@ class Adios(Package):
     """
 
     homepage = "http://www.olcf.ornl.gov/center-projects/adios/"
-    url = "https://github.com/ornladios/ADIOS/archive/v1.9.0.tar.gz"
+    url      = "https://github.com/ornladios/ADIOS/archive/v1.10.0.tar.gz"
+
+    # make -j is currently broken
+    parallel = False
+
+    version('1.10.0', 'eff450a4c0130479417cfd63186957f3')
+    version('1.9.0' , '310ff02388bbaa2b1c1710ee970b5678')
+
+    variant('shared', default=True,
+            description='Builds a shared version of the library')
 
-    version('1.9.0', '310ff02388bbaa2b1c1710ee970b5678')
+    variant('fortran', default=False,
+            description='Enable Fortran bindings support')
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('infiniband', default=False, description='Enable infiniband support')
+
+    variant('zlib', default=True, description='Enable szip transform support')
+    variant('szip', default=False, description='Enable szip transform support')
+    variant('hdf5', default=False, description='Enable HDF5 transport support')
+    variant('netcdf', default=False, description='Enable NetCDF transport support')
 
     # Lots of setting up here for this package
     # module swap PrgEnv-intel PrgEnv-$COMP
     # module load cray-netcdf/4.3.3.1
     # module load cray-hdf5/1.8.14
     # module load python/2.7.10
-    depends_on('hdf5')
-    depends_on('mxml')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('python', type='build')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('mxml@2.9:')
+    # optional transformations
+    depends_on('zlib', when='+zlib')
+    depends_on('szip', when='+szip')
+    # optional transports
+    depends_on('hdf5', when='+hdf5')
+    depends_on('netcdf', when='+netcdf')
+
+    def validate(self, spec):
+        """
+        Checks if incompatible variants have been activated at the same time
+        :param spec: spec of the package
+        :raises RuntimeError: in case of inconsistencies
+        """
+        if '+fortran' in spec and not self.compiler.fc:
+            msg = 'cannot build a fortran variant without a fortran compiler'
+            raise RuntimeError(msg)
 
     def install(self, spec, prefix):
-        configure_args = ["--prefix=%s" % prefix,
-                          "--with-mxml=%s" % spec['mxml'].prefix,
-                          "--with-hdf5=%s" % spec['hdf5'].prefix,
-                          "--with-netcdf=%s" % os.environ["NETCDF_DIR"],
-                          "--with-infiniband=no",
-                          "MPICC=cc", "MPICXX=CC", "MPIFC=ftn",
-                          "CPPFLAGS=-DMPICH_IGNORE_CXX_SEEK"]
```

Comment:
That's a bug in current Spack. To test Adios you need change your packages.yaml to specify defaults for mpi.

> 30 авг. 2016 г., в 0:50, Axel Huebl <notifications@github.com> написал(а):
> 
> In var/spack/repos/builtin/packages/adios/package.py:
> 
> >  
> >      def install(self, spec, prefix):
> > -        configure_args = ["--prefix=%s" % prefix,
> > -                          "--with-mxml=%s" % spec['mxml'].prefix,
> > -                          "--with-hdf5=%s" % spec['hdf5'].prefix,
> > -                          "--with-netcdf=%s" % os.environ["NETCDF_DIR"],
> > -                          "--with-infiniband=no",
> > -                          "MPICC=cc", "MPICXX=CC", "MPIFC=ftn",
> > -                          "CPPFLAGS=-DMPICH_IGNORE_CXX_SEEK"]
> I tried but get an ==> Error: adios does not depend on mpich
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>